modules/packages/conda/metadata.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package conda
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/bzip2"
+	"errors"
+	"io"
+	"strings"
+
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/validation"
+
+	"github.com/klauspost/compress/zstd"
+)
+
+var (
+	ErrInvalidStructure = errors.New("package structure is invalid")
+	ErrInvalidName      = errors.New("package name is invalid")
+	ErrInvalidVersion   = errors.New("package version is invalid")
```

Comment:
It would be helpful to make these wrap the `util.ErrInvalidArgument` error

Reply:
Added this in #22268 for other package types.